### ISSUE TITLE
no-implicit-this and hasBlock

### ISSUE THREAD

**Issue by boris-petrov** [ember-template-lint/ember-template-lint#481]
The `no-implicit-this` rule complains about `hasBlock` but `hasBlock` should not be used with either `this`, nor with `@` (as it returns `undefined`). Perhaps this rule can implicitly ignore `hasBlock`.

**Comment by rwjblue** [comment]
Yep, totally agree! There is a whitelist if built-ins in the rule, and it seems like we should add `hasBlock‘ and `has-block`. Mind sending in a PR?

**Comment by rwjblue** [comment]
Ya, I think `has-block` is generally preferred now (dasherize all the things in hbs files 😝 ), but they both refer to the same thing so it isn't a giant deal...

**Comment by boris-petrov** [comment]
Ah, well, all the JS computed properties are camelCase so definitely not all the things in the HBS files are dasherized. :)

**Comment by sukima** [comment]
How do you enable this pre-beta? This hasn't been released (still in beta since August 30th 2018). Is there a `.template-lintrc.js` fix for this?!

**Comment by sukima** [comment]
The answer for other Google-Fo folks:

```js
  rules: {
    "no-implicit-this": {
      allow: ['has-block', 'hasBlock']
    }
  },
```

**Comment by sukima** [comment]
Also to help others `has-block` **does not work** for me and here is proof: https://ember-twiddle.com/925ee39d84572b41ba94f597e17fca68
